pkg/feature/storage/v2/flag_trigger.go

```diff
@@ -0,0 +1,341 @@
+// Copyright 2023 The Bucketeer Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//go:generate mockgen -source=$GOFILE -package=mock -destination=./mock/$GOFILE
+package v2
+
+import (
+	"context"
+	_ "embed"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/bucketeer-io/bucketeer/pkg/feature/domain"
+	"github.com/bucketeer-io/bucketeer/pkg/storage/v2/mysql"
+	proto "github.com/bucketeer-io/bucketeer/proto/feature"
+)
+
+var (
+	//go:embed sql/flag_trigger/insert_flag_trigger.sql
+	insertFlagTriggerSQL string
+	//go:embed sql/flag_trigger/update_flag_trigger.sql
+	updateFlagTriggerSQL string
+	//go:embed sql/flag_trigger/update_flag_trigger_usage.sql
+	updateFlagTriggerUsageSQL string
+	//go:embed sql/flag_trigger/reset_flag_trigger.sql
+	resetFlagTriggerSQL string
+	//go:embed sql/flag_trigger/delete_flag_trigger.sql
+	deleteFlagTriggerSQL string
+	//go:embed sql/flag_trigger/get_flag_trigger.sql
+	getFlagTriggerSQL string
+	//go:embed sql/flag_trigger/list_flag_trigger.sql
+	listFlagTriggersSQL string
+	//go:embed sql/flag_trigger/count_flag_trigger.sql
+	countFlagTriggersSQL string
+	//go:embed sql/flag_trigger/enable_flag_trigger.sql
+	enableFlagTriggerSQL string
+	//go:embed sql/flag_trigger/disable_flag_trigger.sql
+	disableFlagTriggerSQL string
+)
+
+var (
+	ErrFlagTriggerAlreadyExists          = errors.New("flag trigger: already exists")
+	ErrFlagTriggerNotFound               = errors.New("flag trigger: not found")
+	ErrFlagTriggerUnexpectedAffectedRows = errors.New("flag trigger: unexpected affected rows")
+)
+
+type FlagTriggerStorage interface {
+	CreateFlagTrigger(ctx context.Context, flagTrigger *domain.FlagTrigger) error
+	UpdateFlagTrigger(
+		ctx context.Context,
```

Comment:
To keep the consistency with other places, please pass the `*domain.FlagTrigger` for all the interfaces in the storage.

Reference.
https://github.com/bucketeer-io/bucketeer/blob/main/pkg/autoops/storage/v2/progressive_rollout.go#L38-L56

Also, please use only one update interface.
Reference.
https://github.com/bucketeer-io/bucketeer/blob/main/pkg/experiment/storage/v2/experiment.go#L35-L43